src/actions/tests/pending-breakpoints.js

```diff
@@ -0,0 +1,261 @@
+// TODO: we would like to mock this in the local tests
+const theMockedPendingBreakpoint = {
+  location: {
+    sourceUrl: "http://localhost:8000/examples/bar.js",
+    line: 5,
+    column: undefined
+  },
+  condition: "3",
+  disabled: false
+};
+
+jest.mock("../../utils/prefs", () => ({
+  prefs: {
+    expressions: [],
+    pendingBreakpoints: {
+      "http://localhost:8000/examples/bar.js:5:": {
+        location: {
+          sourceUrl: "http://localhost:8000/examples/bar.js",
+          line: 5,
+          column: undefined
+        },
+        condition: "3",
+        disabled: false
+      }
+    }
+  }
+}));
+
+import {
+  createStore,
+  selectors,
+  actions,
+  makeSource
+} from "../../utils/test-head";
+import {
+  makeLocationId,
+  makePendingLocationId
+} from "../../reducers/breakpoints";
+import expect from "expect.js";
+
+function generateBreakpoint(filename) {
+  return {
+    location: {
+      sourceUrl: `http://localhost:8000/examples/${filename}`,
+      sourceId: filename,
+      line: 5
+    },
+    condition: null,
+    disabled: false
+  };
+}
+
+function generatePendingBreakpoint(breakpoint) {
+  const {
+    location: { sourceUrl, line, column },
+    condition,
+    disabled
+  } = breakpoint;
+
+  return {
+    location: { sourceUrl, line, column },
+    condition,
+    disabled
+  };
+}
+
+function slideMockBp(bp) {
+  const slidBp = Object.assign({}, bp);
+  slidBp.location.line = bp.location.line + 2;
+  return slidBp;
+}
+
+const slidingMockThreadClient = {
+  setBreakpoint: (location, condition) => {
+    return new Promise((resolve, reject) => {
+      const actualLocation = Object.assign({}, location, {
+        line: location.line + 2
+      });
+      resolve({ id: makeLocationId(location), actualLocation, condition });
+    });
+  }
+};
+
+const simpleMockThreadClient = {
+  setBreakpoint: (location, condition) => {
+    return new Promise((resolve, reject) => {
+      resolve({ id: "hi", actualLocation: location });
+    });
+  },
+
+  removeBreakpoint: id => {
+    return new Promise((resolve, reject) => {
+      resolve({ status: "done" });
+    });
+  },
+
+  setBreakpointCondition: (id, location, condition, noSliding) => {
```

Comment:
nitpick: we probably can remove all of the arguments for this method. also the above methods have unused variables